umbral/keys.py

```diff
@@ -108,6 +111,43 @@ def get_pubkey(self):
         """
         return UmbralPublicKey(self.bn_key * self.params.g)
 
+    def to_cryptography_privkey(self):
+        """
+        Returns a cryptography.io EllipticCurvePrivateKey from the Umbral key.
+        """
+        backend = default_backend()
+
+        backend.openssl_assert(self.bn_key.group != backend._ffi.NULL)
+        backend.openssl_assert(self.bn_key.bignum != backend._ffi.NULL)
+
+        ec_key = backend._lib.EC_KEY_new()
+        backend.openssl_assert(ec_key != backend._ffi.NULL)
+        ec_key = backend._ffi.gc(ec_key, backend._lib.EC_KEY_free)
```

Comment:
This pattern is unclean IMO.

Why do we name `ec_key`, only to shadow it two lines later with a different object?  (I notice this happens on two other occasions in this PR).

It reminds me of the similar function in cryptography.io, `_evp_pkey_to_public_key`.  That always bothered me there too.

Can't we do better?

Reply:
No, this is how OpenSSL works. :P

Reply:
Heh, well it doesn't have to be.  Why not change them?

Reply:
Well, an EVP_PKEY is a structure for holding a private key. An EC_KEY essentially like the primitive building block of it, so an EVP_PKEY still requires an EC_KEY.

Reply:
No no, I know that.  I'm talking about the shadowed names.

Reply:
@michwill, @tuxxy, so just to close the loop on the gc discussion, `del` does indeed work (at least for the purposes of the tests):

```
new_ec_key = backend._lib.EC_KEY_new()
backend.openssl_assert(new_ec_key != backend._ffi.NULL)
ec_key = backend._ffi.gc(new_ec_key, backend._lib.EC_KEY_free)
del new_ec_key
```

And while I won't persist in this argument if y'all just want to keep `ec_key` for both, I do think that I like this pattern (and the clarity it provides) better.